x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Verifier.java

```diff
@@ -560,6 +561,37 @@ private static boolean checkGroupMatch(Expression e, Node<?> source, List<Expres
         }
         return false;
     }
+    
+    private static void checkHistogramInGrouping(LogicalPlan p, Set<Failure> localFailures) {
+        // check if the query has a grouping function (Histogram) but no GROUP BY
+        if (p instanceof Project) {
+            Project proj = (Project) p;
+            
+            proj.projections().forEach(e -> {
+                if (Functions.isGrouping(e) == false) {
+                    e.collectFirstChildren(c -> {
+                        if (Functions.isGrouping(c)) {
+                            localFailures.add(fail(c, "[%s] needs to be part of the grouping", Expressions.name(c)));
+                            return true;
+                        }
+                        return false;
+                    });
+                } else {
+                    localFailures.add(fail(e, "[%s] needs to be part of the grouping", Expressions.name(e)));
+                }
+            });
+        } else if (p instanceof Aggregate) {
+            // if it does have a GROUP BY, check if the groupings contain the grouping functions (Histograms) 
+            Aggregate a = (Aggregate) p;
+
+            a.aggregates().forEach(as -> {
+                Expression exp = as instanceof Alias ? ((Alias) as).child() : as;
+                if (Functions.isGrouping(exp) && false == Expressions.anyMatch(a.groupings(), g -> exp.semanticEquals(g))) {
```

Comment:
what if the histogram is wrapped with a Scalar function? is it caught somewhere else?

Reply:
This was already handled. See the test [here](https://github.com/elastic/elasticsearch/blob/f1f73979faaa69de54067437bd3594e779c2e0d0/x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/analysis/analyzer/VerifierErrorMessagesTests.java#L527).